datastore/longhorn.go

```diff
@@ -628,6 +629,11 @@ func CheckVolume(v *longhorn.Volume) error {
 	if v.Name == "" || size == 0 || v.Spec.NumberOfReplicas == 0 {
 		return fmt.Errorf("BUG: missing required field %+v", v)
 	}
+
+	if v.Spec.Encrypted && size <= crypto.Luks2MinimalVolumeSize {
+		return fmt.Errorf("invalid volume size %v, need to be bigger than 16MiB, default LUKS2 header size for encryption", v.Spec.Size)
```

Comment:
nit: you can use `crypto.Luks2MinimalVolumeSize` in log instead of hard-coding `16 MiB`. 

ref: you can use https://pkg.go.dev/github.com/ylin610/units for unit formatting.